ledger/ledger_test.go

```diff
@@ -3010,3 +3019,115 @@ func TestVotersReloadFromDiskPassRecoveryPeriod(t *testing.T) {
+		protocol.StateProofBasic: sp,
+	}
+
+	for i := uint64(0); i < firstStateProofDataConfirmedRound-1; i++ {
+		blk.BlockHeader.Round++
+		blk.BlockHeader.TimeStamp += 10
+		err = l.AddBlock(blk, agreement.Certificate{})
+		require.NoError(t, err)
+	}
+
+	verifyStateProofVerificationTracking(t, &l.stateProofVerification, basics.Round(firstStateProofDataTargetRound),
+		1, proto.StateProofInterval, false, any)
+
+	blk.BlockHeader.Round++
+	blk.BlockHeader.TimeStamp += 10
+	err = l.AddBlock(blk, agreement.Certificate{})
+	require.NoError(t, err)
+
+	verifyStateProofVerificationTracking(t, &l.stateProofVerification, basics.Round(firstStateProofDataTargetRound),
+		1, proto.StateProofInterval, true, trackerMemory)
+
+	for i := firstStateProofDataConfirmedRound; i < lastStateProofDataConfirmedRound; i++ {
+		blk.BlockHeader.Round++
+		blk.BlockHeader.TimeStamp += 10
+		err = l.AddBlock(blk, agreement.Certificate{})
+		require.NoError(t, err)
+	}
+
+	verifyStateProofVerificationTracking(t, &l.stateProofVerification, basics.Round(firstStateProofDataTargetRound),
+		numOfStateProofs-1, proto.StateProofInterval, true, trackerDB)
+	// Last one should be in memory as a result of cfg.MaxAcctLookback not being equal to 0.
+	verifyStateProofVerificationTracking(t, &l.stateProofVerification, basics.Round(lastStateProofDataTargetRound),
+		1, proto.StateProofInterval, true, trackerMemory)
+
+	l.WaitForCommit(blk.BlockHeader.Round)
+
+	verifyStateProofTrackingLedger(t, l, basics.Round(firstStateProofDataTargetRound),
+		numOfStateProofs, proto.StateProofInterval, true)
+
+	var stateProofReceived bookkeeping.StateProofTrackingData
+	stateProofReceived.StateProofNextRound = basics.Round(firstStateProofDataTargetRound + proto.StateProofInterval)
+	blk.BlockHeader.StateProofTracking = map[protocol.StateProofType]bookkeeping.StateProofTrackingData{
+		protocol.StateProofBasic: stateProofReceived,
+	}
+	blk.BlockHeader.Round++
+	blk.BlockHeader.TimeStamp += 10
+
+	// This implementation is an easy way to feed the delta, which the state proof verification tracker relies on,
+	// to the ledger.
+	delta, err := internal.Eval(context.Background(), l, blk, false, l.verifiedTxnCache, nil)
+	delta.StateProofNext = stateProofReceived.StateProofNextRound
+	vb := ledgercore.MakeValidatedBlock(blk, delta)
+	err = l.AddValidatedBlock(vb, agreement.Certificate{})
+
+	require.NoError(t, err)
+
+	for i := uint64(0); i < proto.MaxBalLookback; i++ {
+		blk.BlockHeader.Round++
+		blk.BlockHeader.TimeStamp += 10
+		err = l.AddBlock(blk, agreement.Certificate{})
+		require.NoError(t, err)
+	}
+
+	l.WaitForCommit(blk.BlockHeader.Round)
+
+	verifyStateProofTrackingLedger(t, l, basics.Round(firstStateProofDataTargetRound),
+		1, proto.StateProofInterval, false)
+	verifyStateProofTrackingLedger(t, l, basics.Round(firstStateProofDataTargetRound+proto.StateProofInterval),
+		numOfStateProofs-1, proto.StateProofInterval, true)
+}
```

Comment:
maybe we can add another test that checks the behavior when reloading the new tracker from the disk. i.e make sure that after reloading the ledger the tracker has the same elements in memory.